src/Html/Styled/Lazy.elm

```diff
@@ -78,3 +82,61 @@ lazy6 =
 lazy7 : (a -> b -> c -> d -> e -> f -> g -> Html msg) -> a -> b -> c -> d -> e -> f -> g -> Html msg
 lazy7 =
     VirtualDom.Styled.lazy7
+
+
+{-| Much like `lazy`, but allows specifying a key for the created `Html` node. This can increase performance in cases when you want to lazily generate a list of nodes.
+
+This key becomes the id of the root node and all css that is generated will be scoped to that id. This allows the browser to spend less time calculating styles as there won't be lots styles with the same class name.
+
+Some notes about using this function:
+
+  - The key must be a valid HTML id
```

Comment:
I'm a little surprised by an HTML function with `keyed` in the name using an HTML id. I supposed because [the Html.Keyed](https://package.elm-lang.org/packages/elm/html/latest/Html-Keyed) `node` function _doesn't_ use HTML ids?

I guess I'm wondering if there's another naming pattern that could avoid confusion with Keyed Nodes?

Reply:
I had originally planned on calling this `scoped` and then went with `keyed` after talking with Richard.   Do you `scoped` any better? 

The functions would be `scopedLazy` ... `scopedLazy7` then.

Reply:
Oops missed this comment before, sorry!
I prefer `scoped` -- I think it's a bit more clear